### PR TITLE
Defensive Roll Confirmation Hit Space Bar or timeout

### PR DESCRIPTION
**Pull request by DanVanAtta**
Should address: https://github.com/triplea-game/triplea/issues/537
First commit is a refactor/cleanup.
Second commit updates the logic to still allow for space confirmation before the timeout expires.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/680)

### CODE REVIEW

**Comment by DanVanAtta** [comment]
This is being addressed by #680

File: src/games/strategy/triplea/ui/BattleDisplay.java [hunk 1]
```diff
@@ -303,73 +306,43 @@ public void changedUnitsNotification(final PlayerID player, final Collection<Uni
...
-
-  private void waitForSpaceBarClick(final String message) {
     final CountDownLatch continueLatch = new CountDownLatch(1);
-    // set the action in the swing thread.
-    SwingUtilities.invokeLater(new Runnable() {
-      @Override
-      public void run() {
-        m_actionButton.setAction(new AbstractAction(message) {
-          private static final long serialVersionUID = 4489826259192394858L;
+    SwingUtilities.invokeLater(() -> m_actionButton.setAction(SwingAction.of(message, e -> continueLatch.countDown())));
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/BattleDisplay.java]: Would it be better to break this line up some? Its kind of hard to follow all the nesting in one line.

File: src/games/strategy/triplea/ui/BattleDisplay.java [hunk 2]
```diff
@@ -303,73 +306,43 @@ public void changedUnitsNotification(final PlayerID player, final Collection<Uni
...
+    m_mapPanel.getUIContext().addShutdownLatch(continueLatch);
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/ui/BattleDisplay.java]: Would it be worth renaming all the "m_" member variables in this class now?

**Comment by ron-murhammer** [comment]
@DanVanAtta Added a few comments above. Functionality seems to work well and much better than before now that it can be moved along with spacebar.

File: src/games/strategy/triplea/ui/BattleDisplay.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/BattleDisplay.java]: Broke it up where I saw nesting, in the `setAction(...)` piece: 64c385c

File: src/games/strategy/triplea/ui/BattleDisplay.java (hunk 2 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ui/BattleDisplay.java]: Not quite worth it, let's keep the diff clean and avoid the scope creep for the immediate moment.

File: src/games/strategy/triplea/ui/BattleDisplay.java [hunk 3]
```diff
@@ -303,73 +306,44 @@ public void changedUnitsNotification(final PlayerID player, final Collection<Uni
...
-            continueLatch.countDown();
+    // Set a auto-wait expiration if the option is set.
+    if(!getConfirmDefensiveRolls()) {
+      final int maxWaitTime = 1500;
+      Timer t = new Timer();
+      t.schedule(new TimerTask() {
+        @Override
+        public void run() {
+          continueLatch.countDown();
+          if (continueLatch.getCount() > 0) {
```
Comment (RoiEXLab) [inline src/games/strategy/triplea/ui/BattleDisplay.java]: Just noticed that this is effectively dead code ^^
See #5826
